piecrust/src/imports/wasm32.rs

```diff
@@ -0,0 +1,195 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::sync::Arc;
+
+use dusk_wasmtime::{Caller, Result as WasmtimeResult};
+use piecrust_uplink::{
+    ContractError, ContractId, ARGBUF_LEN, CONTRACT_ID_BYTES,
+};
+
+use crate::imports::{check_arg, check_ptr, POINT_PASS_PCT};
+use crate::instance::Env;
+use crate::Error;
+
+pub(crate) fn hq(
+    mut fenv: Caller<Env>,
+    name_ofs: u32,
```

Comment:
Can you call `wasm64::hq` at this point and cast `name_ofs` to a `u64`? Would save quite some code duplication

Reply:
Let me give these a stab

Reply:
That worked!